src/syscheckd/create_db.c

```diff
@@ -550,7 +559,11 @@ static int fim_resolve_db_collision(unsigned long inode, unsigned long dev) {
 
         w_mutex_lock(&syscheck.fim_entry_mutex);
 
-        switch (fim_process_file_from_db(current_path, stack, tree, &event)) {
+        w_rwlock_rdlock(&syscheck.directories_lock);
+        int ret = fim_process_file_from_db(current_path, stack, tree, &event);
+        w_rwlock_unlock(&syscheck.directories_lock);
```

Comment:
Could we move this locks inside `fim_process_file_from_db`? At this point in the code neither `syscheck.directories` nor `syscheck.wildcards` has been accessed.